external/CMakeLists.txt

```diff
@@ -68,6 +68,8 @@ add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/GSL EXCLUDE_FROM_ALL)
 
 # Add Boost
 add_subdirectory(boost-cmake)
+set(BOOST_INCLUDEDIR ${CMAKE_CURRENT_SOURCE_DIR}/boost-cmake CACHE PATH "Change boost lookup path")
+set(BOOST_LIBRARYDIR ${CMAKE_BINARY_DIR}/boost-cmake CACHE PATH "Change boost lookup path")
```

Comment:
Can I ask why we need these two?

Reply:
Yeah, taily library uses boost but doesn't pull it as submodule, so in order for it to find the one we pull, you need this path. Otherwise, it will either find the system copy (like in my env) or nothing and fail (like on CI).